rpc/src/main/scala/WorkerServer.scala

```diff
@@ -74,30 +90,76 @@ class WorkerServer(executionContext: ExecutionContext) { self =>
       val promise = Promise[PartitionReply]
 
       Future {
-        // TODO: Logic
-        promise.success(new PartitionReply())
+        try {
+          val block = Block.fromPath(Path("data/block"), 10, 90)
+          request.workers
+            .map(workerMetadata => {
+              val keyRange = KeyRange(
+                Key.fromString(workerMetadata.keyRange.get.from.toStringUtf8),
+                Key.fromString(workerMetadata.keyRange.get.to.toStringUtf8)
+              )
```

Comment:
How about something like workerMetadata.KeyRange.get ?

Reply:
For matching the type between `String` and `ByteString`, I use toStringUtf8.